packages/flutter/lib/src/services/asset_bundle.dart

```diff
@@ -72,14 +73,18 @@ class MojoAssetBundle extends AssetBundle {
     _imageCache = null;
   }
 
-  Future<ui.Image> _fetchImage(String key) async {
-    return await decodeImageFromDataPipe(await load(key));
+  Future<ui.Image> _fetchImage(String key, [ AssetResolver resolver ]) async {
```

Comment:
Since you asynchronously resolve the key, even if you have the future for the resolved key already cached, and it's already resolved, the quickest this can get the image on the screen is on frame number 2 of showing a particular image.